### PR TITLE
chore: id mapper as extensions [AR-2886]

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Prepared PoC for migrating Mapper interface to extension functions. Decision for that change was because
- it simplifying their usage, where we don't need to provide interface on each class
- usage of them are more intuitive (we don't need to think which function is proper to our current data class layer
- we don't need to write additional mocks where we are mapping to other models
- it helps to track that we are using other values relative to the corresponding objects of other layers
- it's harder to write duplicated mappers

BEFORE
`
 fun fromApiModel(networkId: NetworkQualifiedId) = QualifiedID(
value = networkId.value, 
domain = networkId.domain
)
`
AFTER
`
fun NetworkQualifiedId.toModel(): QualifiedID = QualifiedID(value, domain)
`

### Issues

- it also fixes differences between test ID models in different layers

### Causes (Optional)
- a lot of code changes, but it can be done incrementally

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by MohamadJaara** [comment]
Can you please fill the template to have an idea about why this change is needed?

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/id/IdMappers.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+fun QualifiedID.toDao(): PersistenceQualifiedId = PersistenceQualifiedId(value, domain)
+
+fun QualifiedIDEntity.toModel(): QualifiedID = QualifiedID(value, domain)
+fun QualifiedIDEntity.toApi(): NetworkQualifiedId = NetworkQualifiedId(value, domain)
+
+fun NetworkQualifiedId.toModel(): QualifiedID = QualifiedID(value, domain)
+fun NetworkQualifiedId.toDao(): PersistenceQualifiedId = PersistenceQualifiedId(value, domain)
+
+fun UserAssetDTO.toDao(domain: String): QualifiedIDEntity = PersistenceQualifiedId(key, domain)
+fun UserAssetDTO.toModel(domain: String): QualifiedID = QualifiedID(key, domain)
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/id/IdMappers.kt:20]: `internal` modifier.

All mapping stuff shouldn't really be exposed to consumer apps.

We should actually go a bit deeper in a future PR and make sure that API models, DB models, etc. also shouldn't be exposed.

Comment (Garzas) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/id/IdMappers.kt:20]: Yup I agree also I think in future we should remove API and DB models from use cases

**Review by vitorhugods — APPROVED** [review]
Nice 👨‍🍳👌

**Review by MohamadJaara — APPROVED** [review]

**Pull request merged**